File: CloudFlow/src/ro/cloudSoft/cloudDoc/presentation/client/shared/model/content/AdminUpdateDocumentModel.java
package ro.cloudSoft.cloudDoc.presentation.client.shared.model.content;

import java.util.List;

public class AdminUpdateDocumentModel {
	
	private String documentLocationRealName;
	private String id;
	private Long documentTypeId;
	private String documentName;
	private String documentDescription;
	private List<MetadataInstanceModel> metadataInstances;
	private List<MetadataCollectionInstanceModel> metadataCollectionInstances;
	
	public String getDocumentLocationRealName() {
		return documentLocationRealName;
	}
	public void setDocumentLocationRealName(String documentLocationRealName) {
		this.documentLocationRealName = documentLocationRealName;
	}
	public String getId() {
		return id;
	}
	public void setId(String id) {
		this.id = id;
	}
	public Long getDocumentTypeId() {
		return documentTypeId;
	}
	public void setDocumentTypeId(Long documentTypeId) {
		this.documentTypeId = documentTypeId;
	}
	public String getDocumentName() {
		return documentName;
	}
	public void setDocumentName(String documentName) {
		this.documentName = documentName;
	}
	public String getDocumentDescription() {
		return documentDescription;
	}
	public void setDocumentDescription(String documentDescription) {
		this.documentDescription = documentDescription;
	}
	public List<MetadataInstanceModel> getMetadataInstances() {
		return metadataInstances;
	}
	public void setMetadataInstances(List<MetadataInstanceModel> metadataInstances) {
		this.metadataInstances = metadataInstances;
	}
	public List<MetadataCollectionInstanceModel> getMetadataCollectionInstances() {
		return metadataCollectionInstances;
	}
	public void setMetadataCollectionInstances(List<MetadataCollectionInstanceModel> metadataCollectionInstances) {
		this.metadataCollectionInstances = metadataCollectionInstances;
	}
}

File: CloudFlow/FrontEnd-App/src/app/shared/model/content/admin-update-document.model.ts
import { JsonObject, JsonProperty } from "json2typescript";
import { MetadataInstanceModel } from "./metadata-instance.model";
import { MetadataCollectionInstanceModel } from "./metadata-collection-instance.model";

@JsonObject
export class AdminUpdateDocumentModel {

	@JsonProperty("documentLocationRealName", String)
	public documentLocationRealName: string = null;

	@JsonProperty("id", String)
	public id: string = null;

	@JsonProperty("documentTypeId", Number)
	public documentTypeId: number = null;	

	@JsonProperty("documentName", String)
	public documentName: string = null;

	@JsonProperty("documentDescription", String, true)
	public documentDescription: string = null;

	@JsonProperty("metadataInstances", [MetadataInstanceModel])
	public metadataInstances: MetadataInstanceModel[] = [];
	
	@JsonProperty("metadataCollectionInstances", [MetadataCollectionInstanceModel], true)
	public metadataCollectionInstances: MetadataCollectionInstanceModel[] = [];
}